x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/pagerduty.tsx

```diff
@@ -355,11 +374,14 @@ const PagerDutyParamsFields: React.FunctionComponent<ActionParamsProps<PagerDuty
               name="timestamp"
               data-test-subj="timestampInput"
               value={timestamp || ''}
+              isInvalid={errors.timestamp.length > 0 && timestamp !== undefined}
               onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                 editAction('timestamp', e.target.value, index);
               }}
               onBlur={() => {
-                if (!timestamp) {
+                if (timestamp?.trim()) {
```

Comment:
I was wondering about doing the trim() in the action params validation rather than in the UI here.  Guessing here fixes it 99% of the time, but are there cases where it could be missed here, and still have leading/trailing whitespace when the validation occurs?  Guessing possible, but very unlikely.  In any case, the validation now prints the date value it's validating with dq's, so ... think this will be user-diagnose-able now where it wasn't before ...

Reply:
Yeah I originally wanted to do it in the action, but that forces us to do trim in multiple places as the validation is separate from where its used and it ends up forcing the `null || string?.trim())` logic to repeat in multiple places unless a more major refactoring was done.